src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/EmptyLineSeparatorCheck.java

```diff
@@ -39,8 +40,7 @@
  * static initializers and instance initializers.
  * </p>
  * <p>
- * ATTENTION: empty line separator is required between AST siblings,
- * not after line where token is found.
+ * ATTENTION: This Check has limited support of comments.
```

Comment:
limited is too cryptic and does not give details.

original ATTENTION note was added by me as result of discussion https://github.com/checkstyle/checkstyle/issues/3089 
and it also not that good as it conflicts with original statement `Checks for empty line separators AFTER header, package ...`

`AST` should be removed to : 
`empty line separator is required between token siblings, not after line where token is found. If token does not have same type sibling then empty line is required at its end (for example for CLASS_DEF it is after '}')`

As you plunged in this Check .... please review issue that referenced above and share your thoughts on is it possible to make Check more clear and predictable on behavior ? It might be better to limit this Check to certain cases and create separate Check to companion to cover other cases.

Reply:
To be honest, I don't have any good suggestions for it.
I do not like such long (and also cryptic :)) description, it is not understandable for user (may be add some examples from test cases).
The problem is that comment policy is not clear (and term 'header' is not defined).
Comments are ignored by default, but for some cases (e.g. one in current issue) it make sense to separate them.
As I said, comment policy should be defined, e.g. there should be no comments on next line after package decl and imports? but trailing comments are ok.

Reply:
> I do not like such long (and also cryptic :)) description, it is not understandable for user (may be add some examples from test cases).

sure in addition, example is always better that numerous words.

> there should be no comments on next line after package decl and imports? but trailing comments are ok.

sounds reasonable.

Reply:
as we have good wording ....
please use what I suggested and add example to show use where empty line is expected for CLASS_DEF.
and I will approve this PR.

> should be no comments on next line after package decl and imports? but trailing comments are ok.

please create separate issue.

Reply:
@romani It is not clear for me what example you mean. Example of violation or allowed code?
And do I understand correctly that it should be in "examples" section?
There is already some examples of violations https://checkstyle.sourceforge.io/apidocs/com/puppycrawl/tools/checkstyle/checks/whitespace/EmptyLineSeparatorCheck.html

Reply:
I try to avoid users questions that triggered such note appearance.

Let's update code example for default config https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/EmptyLineSeparatorCheck.java#L112
To
```
 * class Foo {
 *   public static final int FOO_CONST = 1;
 *
 *   public void foo() {}
 * }
```

To show that no empty like after `class Foo {` is expected.

Reply:
done.